disk.c

```diff
@@ -240,7 +240,8 @@ void diskimage_cachetrack( struct diskimage *dimg, int track, int side )
   while( ptr < eot )
   {
     // Search for ID mark
-    while( (ptr<eot) && (ptr[0]!=0xfe) ) ptr++;
+    while( (ptr<eot) && !((ptr[0]==0xa1) && (ptr[1]==0xa1) && (ptr[2]==0xa1) && (ptr[3]==0xfe)) ) ptr++;
```

Comment:
"ptr<eot" is no longer the correct test if you are checking for 4 bytes.

"((ptr+3)<eot)" should be used

Reply:
Exact, modified